examples/squiral/main.go

```diff
@@ -0,0 +1,371 @@
+// Copyright 2019 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+// This demo is inspired by the xscreensaver 'squirals'.
+
+package main
+
+import (
+	"fmt"
+	"image/color"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+	"golang.org/x/image/colornames"
+)
+
+type direction int
+
+const (
+	width     = 800
+	height    = 600
+	drawScale = 1
+	wwidth    = drawScale * width
+	wheight   = drawScale * height
+	amount    = width / 32
+)
+
+const (
+	east direction = iota
+	south
+	west
+	north
+
+	horizontal direction = iota
```

Comment:
That sounds a little odd to share the same type here. We don't use `horizontal` and `vertical` anyway, right?